romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -0,0 +1,214 @@
+"""
+Module for the multiband source catalog step.
+"""
+
+import logging
+
+import numpy as np
+from astropy.table import Table, join
+from roman_datamodels import datamodels, maker_utils
+
+from romancal.datamodels import ModelLibrary
+from romancal.multiband_catalog.detection_image import make_detection_image
+from romancal.multiband_catalog.utils import update_colnames
+from romancal.source_catalog.background import RomanBackground
+from romancal.source_catalog.detection import make_segmentation_image
+from romancal.source_catalog.source_catalog import RomanSourceCatalog
+from romancal.stpipe import RomanStep
+
+__all__ = ["MultibandCatalogStep"]
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+class MultibandCatalogStep(RomanStep):
+    """
+    Create a multiband catalog of sources including photometry and basic
+    shape measurements.
+
+    Parameters
+    -----------
+    input : str or `ModelLibrary`
+        Path to an ASDF file or a `ModelLibrary`.
+    """
+
+    class_alias = "multiband_catalog"
+    reference_file_types = []
+
+    spec = """
+        bkg_boxsize = integer(default=1000)   # background mesh box size in pixels
+        kernel_fwhms = float_list(default=None)  # Gaussian kernel FWHM in pixels
+        snr_threshold = float(default=3.0)    # per-pixel SNR threshold above the bkg
+        npixels = integer(default=25)         # min number of pixels in source
+        deblend = boolean(default=False)      # deblend sources?
+        aperture_ee1 = integer(default=30)    # aperture encircled energy 1
+        aperture_ee2 = integer(default=50)    # aperture encircled energy 2
+        aperture_ee3 = integer(default=70)    # aperture encircled energy 3
+        ci1_star_threshold = float(default=2.0)  # CI 1 star threshold
+        ci2_star_threshold = float(default=1.8)  # CI 2 star threshold
+        suffix = string(default='cat')        # Default suffix for output files
+        fit_psf = boolean(default=True)      # fit source PSFs for accurate astrometry?
+    """
+
+    def process(self, library):
+        if isinstance(library, str):
+            library = ModelLibrary(library)
+        if not isinstance(library, ModelLibrary):
+            raise TypeError("library input must be a ModelLibrary object")
+
+        # TODO: sensible defaults
+        if self.kernel_fwhms is None:
+            self.kernel_fwhms = [2.0, 20.0]
+
+        # NOTE: I'm assuming here that all of the input images have been
+        # background subtracted, have the same shape, and are pixel
+        # aligned.
+        # TODO: Do we need a separate background subtraction step
+        # prior to this one?
```

Comment:
The incoming L3s have been sky matched but not background subtracted; we probably do want to follow source_catalog and subtract a background.